syntax = "proto2";

import "lib/decoders/decoders.proto";
import "lib/encoders/encoders.proto";
import "lib/imagereader/imagereader.proto";
import "lib/imagewriter/imagewriter.proto";
import "lib/fluxsource/fluxsource.proto";
import "lib/fluxsink/fluxsink.proto";
import "lib/usb/usb.proto";
import "lib/common.proto";

// NEXT_TAG: 14
message ConfigProto {
	optional string comment = 8;
	optional bool is_extension = 13;

	optional ImageReaderProto image_reader = 12;
	optional ImageWriterProto image_writer = 9;
	
	optional FluxSourceProto flux_source = 10;
	optional FluxSinkProto flux_sink = 11;

	optional EncoderProto encoder = 3;
	optional DecoderProto decoder = 4;
	optional UsbProto usb = 5;

	optional RangeProto cylinders = 6;
	optional RangeProto heads = 7;
}

